workers/generic-worker/interactive/interactive.go

```diff
@@ -0,0 +1,217 @@
+// Package interactive provides a way to run an interactive shell in a task.
+package interactive
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os/exec"
+	"sync"
+
+	"github.com/gorilla/websocket"
+)
+
+var upgrader = &websocket.Upgrader{
+	CheckOrigin: func(r *http.Request) bool {
+		return true
```

Comment:
is it something we'll need to have a look in the future? 
Or if it is possible at all to know if origin is allowed to connect here

Reply:
discussed offline, good enough